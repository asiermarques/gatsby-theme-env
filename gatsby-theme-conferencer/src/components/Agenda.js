import React from "react";
import { AgendaSlotType } from "../lib/agendaSlotTypes";

const Agenda = ({ time_slots, tracks, agendaSlotMap }) => (
  <section id="agenda">
    <div className="container">
      <h2 className="text-center">Agenda</h2>
      <div className="table-responsive-sm">
        <table
          id="agenda-table"
          className="table table-striped stacktable large-only"
        >
          <thead>
            <tr>
              <th />
              {tracks &&
                tracks.map((track, index) => (
                  <th key={index} width="40%">
                    {track.name}
                  </th>
                ))}
            </tr>
          </thead>
          <tbody>
            {time_slots &&
              time_slots.map((slot, slot_index) => {
                return (
                  <tr key={slot_index}>
                    <td className="time">{slot}</td>
                    {agendaSlotMap[slot].map((content, index) => {
                      if (content.type === AgendaSlotType.SPEAKER) {
                        return (
                          <td key={index}>
                            <a href={content.content.slug + "#talk"}>
                              <h5>{content.content.talk.title}</h5>
                              <p>{content.content.name}</p>
                            </a>
                          </td>
                        );
                      } else {
                        return (
                          <td key={index}>
                            <p>{content.content}</p>
                          </td>
                        );
                      }
                    })}
                  </tr>
                );
              })}
          </tbody>
        </table>
      </div>
    </div>
  </section>
);
export default Agenda;
