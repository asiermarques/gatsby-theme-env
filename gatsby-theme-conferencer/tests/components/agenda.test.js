import React from "react";
import { describe, it } from "@jest/globals";
import renderer from "react-test-renderer";
import Agenda from "../../src/components/Agenda";
import AgendaConfigFixture from "../__fixtures/AgendaConfig";
import SpeakerFixture from "../__fixtures/Speaker";
import AgendaMapper from "../../src/lib/agendaMapper";

describe("Agenda", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Agenda
          time_slots={AgendaConfigFixture.time_slots}
          tracks={AgendaConfigFixture.tracks}
          agendaSlotMap={AgendaMapper(
            AgendaConfigFixture.time_slots,
            AgendaConfigFixture.tracks,
            [],
            [SpeakerFixture]
          )}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
