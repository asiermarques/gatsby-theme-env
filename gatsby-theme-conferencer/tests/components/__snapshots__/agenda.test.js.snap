// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Agenda renders correctly 1`] = `
<section
  id="agenda"
>
  <div
    className="container"
  >
    <h2
      className="text-center"
    >
      Agenda
    </h2>
    <div
      className="table-responsive-sm"
    >
      <table
        className="table table-striped stacktable large-only"
        id="agenda-table"
      >
        <thead>
          <tr>
            <th />
            <th
              width="40%"
            />
            <th
              width="40%"
            />
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              className="time"
            >
              10:00-11:00
            </td>
            <td>
              <p>
                -
              </p>
            </td>
            <td>
              <p>
                -
              </p>
            </td>
          </tr>
          <tr>
            <td
              className="time"
            >
              11:00-12:00
            </td>
            <td>
              <p>
                -
              </p>
            </td>
            <td>
              <p>
                -
              </p>
            </td>
          </tr>
          <tr>
            <td
              className="time"
            >
              13:00-13:30
            </td>
            <td>
              <p>
                -
              </p>
            </td>
            <td>
              <p>
                -
              </p>
            </td>
          </tr>
          <tr>
            <td
              className="time"
            >
              13:30-15:00
            </td>
            <td>
              <p>
                -
              </p>
            </td>
            <td>
              <p>
                -
              </p>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</section>
`;
